python_modules/libraries/dagster-aws/dagster_aws/ecs/tasks.py

```diff
@@ -59,10 +59,10 @@ def from_task_definition_dict(task_definition_dict, container_name):
             sidecars=sidecars,
         )
 
-    def task_definition_dict(self):
+    def task_definition_dict(self, launch_type):
         kwargs = dict(
             family=self.family,
-            requiresCompatibilities=["FARGATE"],
+            requiresCompatibilities=["FARGATE"] if launch_type == "FARGATE" else [],
```

Comment:
So this can truly be empty if it's EC2? I see in the API docs its optional but presumably it doesn't work on Fargate without passing `["FARGATE"]`. Have we proven it works on EC2 without passing `["EC2"]`?

https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RegisterTaskDefinition.html#API_RegisterTaskDefinition_RequestSyntax